api/cmd/portainer/main.go

```diff
@@ -70,6 +71,15 @@ func initDataStore(flags *portainer.CLIFlags, fileService portainer.FileService,
 	if err != nil {
 		panic(err)
 	}
+
+	if bconn, ok := connection.(*boltdb.DbConnection); ok {
+		bconn.MaxBatchSize = *flags.MaxBatchSize
+		bconn.MaxBatchDelay = *flags.MaxBatchDelay
+		bconn.InitialMmapSize = *flags.InitialMmapSize
+	} else {
+		panic("unexpected database type")
```

Comment:
any reason why using a panic vs `log.Fatal`?

Reply:
Just mimicking the previous error https://github.com/portainer/portainer/pull/6389/files/160cf2dd9680defa1d18e3094645a5f236bf27f3#diff-2f70d50e7cfe87b317ab74d6141a3aa9e07cb8de9c740c61c247eca17233d997L71

But I can change it if you think it's better.

Reply:
Except right above this place, perhaps I should change both instances then?

Reply:
sounds good. Can you also add to the message what failed? so if a user sees the message we know where to look for it?